lib/elixir/lib/map.ex

```diff
@@ -93,11 +95,18 @@ defmodule Map do
   """
   @spec new(Enum.t, (term -> {key, value})) :: map
   def new(enumerable, transform) do
-    fun = fn el, acc ->
-      {k, v} = transform.(el)
-      put(acc, k, v)
-    end
-    Enum.reduce(enumerable, %{}, fun)
+    enumerable
+    |> Enum.to_list
```

Comment:
Ah so Enum.to_list does a reverse already? That makes sense. I'll run the benchmark again but IIRC manual recursion over Enum.reduce got us 20%. I'm fine if that isn't worth it in this case, just thought I'd make a note.